compiler/src/iree/compiler/Dialect/HAL/Target/ROCM/CMakeLists.txt

```diff
@@ -49,35 +49,75 @@ iree_cc_library(
 
 ### BAZEL_TO_CMAKE_PRESERVES_ALL_CONTENT_BELOW_THIS_LINE ###
 
-# Query for list of device libs to build.
-get_property(AMD_DEVICE_LIBS GLOBAL PROPERTY AMD_DEVICE_LIBS)
-set(AMD_REQUIRED_LIBS "ocml" "ockl" "opencl")
+# We either use device bitcode files from a configurable path
+# or we download them from a fixed location and copy them from
+# there. We default to downloading as it is a better use experience
+# by default.
+# See: https://github.com/shark-infra/amdgpu-device-libs
+set(_amd_required_libs "ocml.bc" "ockl.bc" "opencl.bc")
+set(_amd_device_bc_url 
+    "https://github.com/shark-infra/amdgpu-device-libs/releases/download/v20231101/amdgpu-device-libs-llvm-6086c272a3a59eb0b6b79dcbe00486bf4461856a.tgz")
+set(_amd_device_bc_sha256 "336362416c68fdd8bb80328f65ca7ebaa0c119ea19c95df6df30c832a4df39b9")
+set(_amd_device_bc_stamp "${_amd_device_bc_url} : ${_amd_device_bc_sha256}")
+set(IREE_TARGET_BACKEND_ROCM_DEVICE_BC_PATH "" CACHE PATH 
+    "Optional path to load device bitcode from (default to dynamic fetch)")
+set(_platform_lib_srcdir)
 
-set(_platform_lib_reldir "iree_platform_libs/rocm")
-file(MAKE_DIRECTORY "${IREE_COMPILER_DYLIB_DIR}/${_platform_lib_reldir}")
+if(IREE_TARGET_BACKEND_ROCM_DEVICE_BC_PATH)
+  # Don't fetch: Get from local directory.
+  set(_platform_lib_srcdir "${IREE_TARGET_BACKEND_ROCM_DEVICE_BC_PATH}")
+else()
+  # Fetch from remote archive.
+  set(_platform_lib_archive "${CMAKE_CURRENT_BINARY_DIR}/_fetch_device_libs.tgz")
+  set(_platform_lib_srcdir "${CMAKE_CURRENT_BINARY_DIR}/_fetch_device_libs")
+  set(_fetch_stamp_file "${CMAKE_CURRENT_BINARY_DIR}/_fetch_device_libs.stamp")
+  set(_needs_fetch ON)
+  if(EXISTS "${_fetch_stamp_file}" AND IS_DIRECTORY "${_platform_lib_srcdir}")
+    file(READ "${_fetch_stamp_file}" _stamp_contents)
+    if ("${_stamp_contents}" STREQUAL "${_amd_device_bc_stamp}")
+      set(_needs_fetch OFF)
+    endif()
+  endif()
+
+  # Download, extract, stamp.
+  if(_needs_fetch)
+    message(STATUS "Downloading AMD Device bitcode from ${_amd_device_bc_url} to ${_platform_lib_srcdir}")
+    file(DOWNLOAD "${_amd_device_bc_url}" "${_platform_lib_archive}"
+      EXPECTED_HASH SHA256=${_amd_device_bc_sha256})
```

Comment:
CI logs indicate that this is fast (< 1 second): https://github.com/openxla/iree/actions/runs/6725312997/job/18279438918?pr=15374#step:4:538

(file is less than 1MB)

Reply:
Yeah, it's like 600KiB.